XamlControlsGallery/Navigation/NavigationRootPage.xaml.cs

```diff
@@ -424,14 +425,83 @@ private void OnControlsSearchBoxTextChanged(AutoSuggestBox sender, AutoSuggestBo
 
         private void OnControlsSearchBoxQuerySubmitted(AutoSuggestBox sender, AutoSuggestBoxQuerySubmittedEventArgs args)
         {
+            NavigationRootPage navigationRootPage = NavigationRootPage.GetForElement(this);
             if (args.ChosenSuggestion != null && args.ChosenSuggestion is ControlInfoDataItem)
             {
-                var itemId = (args.ChosenSuggestion as ControlInfoDataItem).UniqueId;
-                Navigate(typeof(ItemPage), itemId);
+                var infoDataItem = args.ChosenSuggestion as ControlInfoDataItem;
+                var itemId = infoDataItem.UniqueId;
+                EnsureItemIsVisibleInNavigation(infoDataItem.Title);
+                navigationRootPage.RootFrame.Navigate(typeof(ItemPage), itemId);
             }
             else if (!string.IsNullOrEmpty(args.QueryText))
             {
-                Navigate(typeof(SearchResultsPage), args.QueryText);
+                navigationRootPage.RootFrame.Navigate(typeof(SearchResultsPage), args.QueryText);
+            }
+        }
+
+        public void EnsureItemIsVisibleInNavigation(string name)

```

Comment:
Was this backported from the WinUI 2 version or should it be ported to the WinUI 2 version?

Reply:
This PR is a sync of changes that were made internally on WinUI3 with the purpose of unforking and removing the internal copy. If changes need to be made in either WinUI3 or in WinUI2, please file an issue and we will fix that after the sync is complete.  Thanks!